assets/svelte/components/SelectedElementFloatingMenu.svelte

```diff
@@ -0,0 +1,258 @@
+      let {currentIndex, destinationIndex} = findSwappedIndexes(dragDirection, e);
+      if (currentIndex === destinationIndex) {
+        // No drag, reset effect.
+        if (newIndex !== null) {
+          // Reset the transforms on all elements but the one being dragged, which must keep following the cursor.
+          Array.from($dragElementInfo.parentElementClone.children).forEach((el, i) => i !== destinationIndex && (el.style.transform = null));    
+        }
+        newIndex = null;
+        calculatePlaceholderPosition(dragDirection, currentIndex, destinationIndex, $dragElementInfo.siblingLocationInfos);
+      } else {
+        let rearrangedInfos = sortedLocationInfos($dragElementInfo.siblingLocationInfos, currentIndex, destinationIndex);
+        repositionGhosts(dragDirection, currentIndex, destinationIndex, rearrangedInfos);
+        calculatePlaceholderPosition(dragDirection, currentIndex, destinationIndex, rearrangedInfos);
+        newIndex = destinationIndex;
+      }
+    }     
+  }
+
+  function handleMousemove(e: MouseEvent) {
+    let ghostElement = getGhostElement();
+    let dragDirection = getDragDirection(ghostElement);
+    let mouseDiff: Coords = {
+      x: e.x - mouseDownEvent.x,
+      y: e.y - mouseDownEvent.y,
+    }
+    if (dragDirection === 'vertical') {
+      dragHandleElement.style.transform = `translateY(${mouseDiff.y}px)`;
+      ghostElement.style.transform = `translateY(${mouseDiff.y}px)`;
+    } else {
+      dragHandleElement.style.transform = `translateX(${mouseDiff.x}px)`;
+      ghostElement.style.transform = `translateX(${mouseDiff.x}px)`;      
+    }
+
+    updateSiblingsPositioning(dragDirection, mouseDiff, e);
+  }
+</script>
+
+{#if $selectedElementMenu}
+  {#if placeholderStyle}
+    <div 
+      class="absolute transition-all" 
+      style="background-color:aqua; opacity: 0.5; {placeholderStyle}"></div>
+  {/if}
+  <button 
+    bind:this={dragHandleElement}
+    on:mousedown={handleMousedown}
+    class="rounded-full w-6 h-6 flex justify-center items-center absolute bg-blue-600 text-white hover:bg-blue-700 focus:outline-none focus-visible:ring-4 focus-visible:ring-blue-200 active:bg-blue-800"
+    class:pointer-events-none={$selectedElementMenu.dragging}
+    class:rotate-90={$selectedElementMenu.dragDirection === 'horizontal'}
+    style={$dragHandleStyle}>
+    <svg xmlns="http://www.w3.org/2000/svg" width="12" height="12" ><path d="M 1 2.5 C 1 1.948 1.448 1.5 2 1.5 L 10 1.5 C 10.552 1.5 11 1.948 11 2.5 L 11 2.5 C 11 3.052 10.552 3.5 10 3.5 L 2 3.5 C 1.448 3.5 1 3.052 1 2.5 Z" fill="currentColor"></path><path d="M 1 6 C 1 5.448 1.448 5 2 5 L 10 5 C 10.552 5 11 5.448 11 6 L 11 6 C 11 6.552 10.552 7 10 7 L 2 7 C 1.448 7 1 6.552 1 6 Z" fill="currentColor"></path><path d="M 1 9.5 C 1 8.948 1.448 8.5 2 8.5 L 10 8.5 C 10.552 8.5 11 8.948 11 9.5 L 11 9.5 C 11 10.052 10.552 10.5 10 10.5 L 2 10.5 C 1.448 10.5 1 10.052 1 9.5 Z" fill="currentColor"></path></svg>        
+  </button>
```

Comment:
Is the icon we're using chosen on purpose, or did we copy it from other apps? For me it wasn't intuitive right away, because for me it looked like a menu, instead of a button to "grap" and move. Maybe an arrow icon like  would make more sense? Although I'll let this choice to any designers or UXD people on the team.